src/metabase/models/setting.clj

```diff
@@ -435,50 +505,53 @@
-                                              true  "true"
-                                              false "false"
-                                              nil   nil))))
-
-(defn set-integer!
-  "Set the value of integer `setting-definition-or-name`."
-  [setting-definition-or-name new-value]
-  (set-string! setting-definition-or-name (when new-value
-                                            (assert (or (integer? new-value)
-                                                        (and (string? new-value)
-                                                             (re-matches #"^\d+$" new-value))))
-                                            (str new-value))))
-
-(defn set-double!
-  "Set the value of double `setting-definition-or-name`."
-  [setting-definition-or-name new-value]
-  (set-string! setting-definition-or-name (when new-value
-                                            (assert (or (number? new-value)
-                                                        (and (string? new-value)
-                                                             (re-matches #"[+-]?([0-9]*[.])?[0-9]+" new-value))))
-                                            (str new-value))))
-
-(defn set-json!
-  "Serialize `new-value` for `setting-definition-or-name` as a JSON string and save it."
-  {:style/indent 1}
-  [setting-definition-or-name new-value]
-  (set-string! setting-definition-or-name (some-> new-value json/generate-string)))
-
-(defn set-timestamp!
-  "Serialize `new-value` for `setting-definition-or-name` as a ISO 8601-encoded timestamp string and save it."
-  [setting-definition-or-name new-value]
-  (set-string! setting-definition-or-name (some-> new-value u.date/format)))
+(defmethod set-value-of-type! :keyword
+  [_setting-type setting-definition-or-name new-value]
+  (set-value-of-type!
+   :string setting-definition-or-name
+   (u/qualified-name new-value)))
+
+(defmethod set-value-of-type! :boolean
+  [setting-type setting-definition-or-name new-value]
+  (if (string? new-value)
+    (set-value-of-type! setting-type setting-definition-or-name (string->boolean new-value))
+    (let [s (case new-value
+              true  "true"
+              false "false"
+              nil   nil)]
+      (set-value-of-type! :string setting-definition-or-name s))))
+
+(defmethod set-value-of-type! :integer
+  [_setting-type setting-definition-or-name new-value]
+  (set-value-of-type!
+   :string setting-definition-or-name
+   (when new-value
+     (assert (or (integer? new-value)
+                 (and (string? new-value)
+                      (re-matches #"^\d+$" new-value))))
+     (str new-value))))
+
+(defmethod set-value-of-type! :double
+  [_setting-type setting-definition-or-name new-value]
+  (set-value-of-type!
+   :string setting-definition-or-name
+   (when new-value
+     (assert (or (number? new-value)
+                 (and (string? new-value)
+                      (re-matches #"[+-]?([0-9]*[.])?[0-9]+" new-value))))
```

Comment:
what about +- infty, nan, etc?

Reply:
Good question. I don't think we want people setting something like `max-rows-limit` to infinity anyway. At any rate this isn't actually new code it's just moved around and is what we already had